enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,353 @@
+our customer use cases.
+This enhancement introduces a new node-plugin that operates at runtime-level
+for extending CPU resources management on top of Kubernetes and OpenShift platforms.
+With the existing CPU management design, a container can either request exclusive CPUs or shared CPUs,
+while with this new node plugin, it would be possible for container workload to request for both.
+
+## Motivation
+
+There is a growing interest from telco users in Kubernetes/OpenShift as a cloud management platform.
+This proposal is motivated by the desire of optimizing the platform's capabilities for telco users,
+such as cost reduction, power efficiency, higher pod density and sustainability.
+
+The telco users have new use cases and requirements that especially related to resource management of the platform. 
+In order to address some of those needs,
+Kubelet CPU manager was created to allow users requesting for exclusive, isolated cpus.
+
+But there are additional use cases that require more granular control of CPU management.
+This enhancement aims to solve the following use case:
+When the CPU manager allocates an exclusive set of CPUs for a given workloads, the workload shall still have the option 
+to specify and access a set of shared CPUs, whether those shared CPUs are shared among containers in the 
+same pod or containers in different ones.
+
+By giving the users the flexibility to use exclusive CPUs only when it is an absolute necessity, we help them utilize
+their existing compute resources in an optimal manner.
+
+### User Stories
+
+* As a developer who's building DPDK application,
+I want to run my application in a container,
+giving full exclusive access to cpu cores for the busy-loops polling threads,
+but having the light-weight tasks like configuration processing and log printing not taking full cores,
+in order to improve the application density and make more efficient use of the CPUs.
+For that, my application container needs access to both shared and exclusive CPUs.
+
+  
+### Goals 
+* Increase pod density, power efficiency, cost reduction and sustainability by optimizing CPUs resource management. 
+* Provide a special-purpose at a runtime-level solution for containers in a pod to have the ability
+to request and subsequently be allocated both exclusive and shared CPUs.
+* Enable the telco-specific use case with a self-contained change,
+    minimizing the changes to the platform
+
+### Non-Goals
+* Introducing a generic mechanism in the platform that does involve Kubelet and pod spec changes.
+
+## Proposal
+
+Container processes are restricted
+to given [cpuset](https://man7.org/linux/man-pages/man7/cpuset.7.html) by their corresponding cgroup definition.
+Essentially, the kernel enforces the processes of container to run only at the CPU (core) ids specified under its
+cgroup.
+
+The mixed-cpu-node-plugin is a  runtime-agnostic, hybrid (composed of both [NRI-plugin](https://github.com/containerd/nri) and
```

Comment:
I'm a big fan of using NRI for this work. However, since we've not converted the existing high perf hooks to NRI, I think there's the possibility of the two hook mechanisms to step on each others toes. thoughts?

Reply:
Multiple NRI plugins can be called sequentially for the same container.
So, those could be a separate NRI plugins that works in parallel.

We need to carefully review the high-perf NRI plugin with this feature in mind, but I don't see any reason why they cannot work together.